lib/wibox/widget/imagebox.lua

```diff
@@ -79,36 +81,48 @@ end
 --- Set an imagebox' image
 -- @property image
 -- @param image Either a string or a cairo image surface. A string is
---   interpreted as the path to a png image file.
+--   interpreted as the path to a image file.
 -- @return true on success, false if the image cannot be used
-
 function imagebox:set_image(image)
```

Comment:
I get a bit lost in this function. I feel like it could help to factor out some helper functions from this, do you agree? (But only those that would be called at least twice, but my feeling is that both of these would be called at least twice)
```
local function set_surface(self, surf)
   self._private.default_size = { width = surf:get_width(), height = surf:get_height() } -- Does it make sense to rename .default like this?
   self._private. handle = nil
   self._private.image = surf
end
local function set_handle(self, handle)
    local dim = handle:get_dimensions()
    self._private.default_size = { width = dim.width, height = dim.height }
    self._private.handle = handle
    self._private.image = nil
end
```
(Sorry that it took me so long to look at this)

Reply:
Can't see why`set_handle` should be called more than once. I personally don't mind single use helpers btw.

Reply:
Should `:set_image` work when called with an RsvgHandle, just like it can already be called with a cairo surface? That would mean there is another call...

Reply:
Sorry, I was not consider RsvgHandle as `set_image` argument. Was thought it will be purely imagebox inner object.

Just to confirm: you are meaning `string|surface|handle|nil`all the bunch of types should be allowed for `imagebox:set_image`?

Reply:
Kind of. I am asking if it makes sense to allow that, or not. So, basically yes, that's what I mean.